docs/readme-zephyr.md

```diff
@@ -156,3 +156,56 @@ although MCUboot's key management infrastructure supports multiple keypairs.
 
 Once MCUboot is built, this new keypair file (`mykey.pem` in this
 example) can be used to sign images.
+
+## Serial recovery
+
+### Interface selection
+
+A serial recovery protocol is available over either a serial port or an USB CDC ACM virtual serial port.
+The SMP server implementation can be enabled by ``CONFIG_MCUBOOT_SERIAL=y``.
+To set a type of an interface, use ``BOOT_SERIAL_DEVICE`` Kconfig choice, and select either ``CONFIG_BOOT_SERIAL_UART`` or ``CONFIG_BOOT_SERIAL_CDC_ACM`` value.
+Which interface belongs to the protocol shall be set by the devicetree-chosen node:
+- `zephyr,console` if serial port used.
+- `zephyr,cdc-acm-uart` if virtual serial port used.
+
+### Entering to the serial recovery mode
+
+To enter the serial recovery mode, the device has to initiate rebooting, and a triggering event has to occur, for example, pressing a button.
+
+By default, the serial recovery GPIO pin active state desires entering the mode.
+The pin shall be assigned using the ``mcuboot_button0`` devicetree button alias.
+
+Alternatively, MCUboot can wait for the prescribed duration to see if DFU is invoked by receiving an MCUmgr command.
+Select ``CONFIG_BOOT_SERIAL_WAIT_FOR_DFU=y`` to use this mode. ``CONFIG_BOOT_SERIAL_WAIT_FOR_DFU_TIMEOUT`` option defines
+the amount of time the device will wait for the trigger.
+
+### Direct image upload
+
+The SPM server implementation allows to select image number for DFU when invoking the MCUmgr `image upload` command.
```

Comment:
```suggestion
By default, the SMP server implementation will only use the the first slot, this can be changed to allow selecting the image number for DFU when invoking the MCUmgr `image upload` command when ``CONFIG_MCUBOOT_SERIAL_DIRECT_IMAGE_UPLOAD=y`` is enabled (note that multiple images can be supported by adjusting ``CONFIG_UPDATEABLE_IMAGE_NUMBER``).
```